### PR TITLE
feat: Add native ESM support

### LINKED ISSUE #85: Native ESM (Node 13+) Support
## Issue Type

- [ ] Bug Report
- [x] Feature Request
- [ ] Other

## Expected

ESM support so we can do this:

```js
import cac from 'cac'

const cli = cac()

// ...
```

## Actual

No ESM support for Node.js

## Possible Solutions

Adding these fields to package.json

```json
"type": "module",
"module: "...",
"exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.cjs"
    },
    "./package.json": "./package.json",
    "./": "./"
  }
```

Note that CJS will still work as before, here's an example of such module: https://github.com/talentlessguy/es-mime-types

Because this project already uses Rollup, it will be very easy to add a module target.

## Info

- CAC version: 6.15.3

if needed, I can make a pull request

### PR DESCRIPTION
**Pull request by talentlessguy**
Add required fields for proper ESM support, closes #85

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,16 @@
...
   "main": "dist/index.js",
+  "module": "dist/mod.js",
   "types": "dist/index.d.ts",
+  "exports": {
+    ".": {
+      "import": "./dist/mod.js",
+      "require": "./dist/index.js"
+    },
+    "./package.json": "./package.json",
+    "./": "./"
```
Comment (egoist) [inline package.json:18]: why is this needed 🤔

```json
    "./package.json": "./package.json",
    "./": "./"
```

Comment (egoist) [inline package.json:18]: why is this needed 🤔

```json
    "./package.json": "./package.json",
    "./": "./"
```

Comment (talentlessguy) [reply package.json:18]: So `mod/package.json` and file in the `mod/*` will resolve properly

can be removed though, not 100% necessary
